fastlane/Helpers/Symbol.swift

```diff
@@ -10,18 +10,58 @@ import Foundation
 
 enum Symbol {
 
-    static func uploadToCrashlytics(environment: Constant.Environment) {
-        let buildURL = URL(fileURLWithPath: Constant.outputPath)
-        let dsymURL = buildURL.appendingPathComponent(environment.productName + ".app.dSYM.zip")
-        guard FileManager.default.fileExists(atPath: dsymURL.relativePath) else {
+    static func uploadAdhocToCrashlytics(environment: Constant.Environment) {
+        guard FileManager.default.fileExists(atPath: environment.dsymPath) else {
             return log(message: "Can't find the dSYM file")
         }
         uploadSymbolsToCrashlytics(
-            dsymPath: dsymURL.relativePath,
+            dsymPath: environment.dsymPath,
             gspPath: .userDefined(environment.gspPath),
             appId: .userDefined(environment.firebaseAppId),
             binaryPath: .userDefined(Constant.uploadSymbolsBinaryPath),
             debug: true // We sometimes has issues with dSYM files, so I enabled this flag.
         )
     }
+
+    static func uploadAppStoreToCrashlytics(
+        versionNumber: String,
+        buildNumber: String,
+        environment: Constant.Environment = .production
+    ) {
+        // This file name from download_dsyms action
+        // https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/download_dsyms.rb#L183
+        let dsymFileName = "\(environment.bundleId)-\(versionNumber)-\(buildNumber).dSYM.zip"
+        let outputDirectoryURL = URL(fileURLWithPath: Constant.outputPath)
+        let dsymPath = outputDirectoryURL.appendingPathComponent(dsymFileName).relativePath
+        guard FileManager.default.fileExists(atPath: dsymPath) else {
+            return log(message: "Can't find the dSYM file")
+        }
+        uploadSymbolsToCrashlytics(
+            dsymPath: dsymPath,
+            gspPath: .userDefined(environment.gspPath),
+            appId: .userDefined(environment.firebaseAppId),
+            binaryPath: .userDefined(Constant.uploadSymbolsBinaryPath),
+            debug: true // We sometimes has issues with dSYM files, so I enabled this flag.
+        )
+    }
+
+    static func downloadFromAppStore(
+        versionNumber: String,
+        buildNumber: String,
+        environment: Constant.Environment = .production
+    ) {
+        // Create output directory if needed
+        let outputDirectoryURL = URL(fileURLWithPath: Constant.outputPath)
+        try? FileManager.default.createDirectory(atPath: outputDirectoryURL.relativePath, withIntermediateDirectories: true)
```

Comment:
Should we handle if this `try?` fails e.g. logging error message for later debugging instead of just ignoring it? 🤔

Reply:
Fixed in https://github.com/nimblehq/ios-templates/pull/361/commits/c654fb72bf5aa6f942520f73fb7b391b953a3e58